lib/src/multi_channel.dart

```diff
@@ -141,7 +141,7 @@ class _MultiChannel<T> extends StreamChannelMixin<T>
         onDone: () => _closeChannel(0, 0));
 
     _innerStreamSubscription = _inner!.stream.cast<List>().listen((message) {
-      var id = message[0] as int;
+      var id = (message[0] as num).toInt();
```

Comment:
I hope this is "free" when it IS an `int`

Reply:
It is: https://github.com/dart-lang/sdk/blob/4eda3397f392ba806fa1e79264767e36a20d26de/sdk/lib/_internal/vm/lib/integers.dart#L310

Reply:
Yea, I would expect this to be at most just the overhead of a function call on all backends.